front/src/components/drive/FileList.tsx

```diff
@@ -209,89 +210,85 @@ const FileList: React.FC<FileListProps> = ({ files, folders, actions }) => {
-									<FileCard
-										metadata={file}
-										key={file.path}
-										selected={selectedItems.has(file.path)}
-										onLeftClick={() => handleLeftClick(file.path)}
-										onRename={actions.has("rename") ? () => handleRename(file.path, false) : undefined}
-										onDelete={actions.has("delete") ? () => handleDelete(file.path, false) : undefined}
-										onMove={actions.has("move") ? () => handleMove(file.path, false) : undefined}
-										onDownload={actions.has("download") ? () => handleDownloadFile(file) : undefined}
-										onRestore={actions.has("restore") ? () => handleRestoreFile(file.path) : undefined}
-									/>
-								))}
+							<div onClick={() => handleSort("size")} className="cursor-pointer">
+								Size {sortColumn === "size" && (sortOrder === "asc" ? "↑" : "↓")}
+							</div>
+							<div onClick={() => handleSort("created_at")} className="cursor-pointer">
+								Created At {sortColumn === "created_at" && (sortOrder === "asc" ? "↑" : "↓")}
 							</div>
 						</div>
+
+						<div className="overflow-y-auto">
+							{currentWorkingDirectory !== "/" && (
+								<FileCard
+									folder
+									metadata={{ path: "..", created_at: new Date().toISOString(), deleted_at: null }}
+									onLeftClick={() =>
+										setCurrentWorkingDirectory(currentWorkingDirectory.split("/").slice(0, -1).join("/") || "/")
+									}
+								/>
+							)}
+
+							{/*TODO: concat both lists and show them together, so it can be sorted by name*/}
+							{folders.map((folder) => (
+								<FileCard
+									metadata={folder}
+									folder
+									key={folder.path}
+									selected={selectedItems.has(folder.path)}
+									onLeftClick={() => handleGoToDirectory(folder.path)}
+									onRename={actions.has("rename") ? () => handleRename(folder.path, true) : undefined}
+									onDelete={actions.has("delete") ? () => handleDelete(folder.path, true) : undefined}
+									onMove={actions.has("move") ? () => handleMove(folder.path, true) : undefined}
+								/>
+							))}
+
+							{files.map((file) => (
+								<FileCard
+									metadata={file}
+									key={file.path}
+									selected={selectedItems.has(file.path)}
+									onLeftClick={() => handleLeftClick(file.path)}
+									onRename={actions.has("rename") ? () => handleRename(file.path, false) : undefined}
+									onDelete={actions.has("delete") ? () => handleDelete(file.path, false) : undefined}
+									onMove={actions.has("move") ? () => handleMove(file.path, false) : undefined}
+									onDownload={actions.has("download") ? () => handleDownloadFile(file) : undefined}
+									onRestore={actions.has("restore") ? () => handleRestoreFile(file.path) : undefined}
+								/>
+							))}
+						</div>
 					</div>
-				</>
-			) : (
-				<div className="flex items-center justify-center h-screen">
-					{/*TODO: add a real loader*/}
-					<div className="text-2xl font-semibold">Loading...</div>
 				</div>
-			)}
```

Comment:
why no more loading ?

Reply:
Moved it at the top of the function (early return) to avoid doing the big ternary here